addons/stock/report/report_stock_reception.py

```diff
@@ -66,6 +66,16 @@ def _get_report_values(self, docids, data=None):
                         # if immediate transfer is not Done and quantity_done hasn't been edited, then move.product_qty will incorrectly = 1 (due to default)
                         quantity_to_assign = move.product_uom._compute_quantity(move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
                     product_to_qty_to_assign[move.product_id].append((quantity_to_assign - qty_already_assigned, move))
+            # On immediate transfers, cover move.product_qty == qty_already_assigned == 0
+            elif move.product_qty == 0:
```

Comment:
Immediate transfers are already handle three lines above, I think it's better to change the following condition
```python
if move.product_qty != qty_already_assigned:
```
into something like
```python
if move.product_qty != qty_already_assigned or (move.product_qty == 0 and move.quantity != qty_already_assigned):
```

Reply:
Doesn't it make a bit less sense to merge the conditions when on the one I've added, we have to use the move line vals vs. the existing one which uses the move vals? If we OR them together then we would just need more redundant checks in that existing block

Reply:
Sorry, a few comments:

- Steve is partially correct, you should add in the condition as he suggests + adjust all the things below it to match

- The `move.product_qty will incorrectly = 1 (due to default)` comment is no longer valid (it was the case in v16 and earlier versions) + the code under it. You should adjust this to correctly grab the draft immediate transfer quantities

- In order to keep the calculations/logic correct, you'll also have to adjust the `if move.move_dest_ids:` condition to consider that the `qty_already_assigned` might not be based on the `move.product_qty` (i.e. is based on the `move.quantity`). This is part of what you'll have to adjust when you apply point 1.

- All that being said, maybe we should only consider using the quantity amount being assignable when the moves are done? This may also fix behavior for MO allocation reports too since the `product_qty` (Qty to produce) used to update to match `qty_producing` (Qty Produced) in v16, but doesn't in v17 so the quantity available to allocate only matches the original qty to produce...

- The manually created MLs should auto-create move(s) once they're saved and the move auto-adds their quantity together, so you should never have to directly reference the MLs except in the assign/unassign methods (which I hopefully made robust enough to still work even in v17....)